### PR TITLE
 Chinese - Payroll

### PR DESCRIPTION
**Pull request by xie8899**
[UPD] add a new module, Chinese - Payroll

### CODE REVIEW

**Comment by xie8899** [comment]
cc @oca-clabot @elicoidal

**Comment by xie8899** [comment]
cc @elicoidal @duanyp1991

**Comment by elicoidal** [comment]
LGTM

File: l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    social_insurance_amount = fields.Float(
+        'Social Insurance Base',
+        digits_compute=dp.get_precision('Payroll')
+    )
+    pit_base_amount = fields.Float(
+        'PIT Base',
+        digits_compute=dp.get_precision('Payroll'),
+        default=0
+    )
+    pit_exemption_amount = fields.Float(
```
Comment (elicoidal) [inline l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py:27]: Please remove this field

File: l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        'Individual Income Tax Threshold',
+        required=True,
+        digits_compute=dp.get_precision('Payroll'),
+        default=3500.00
+    )
+
+    @api.one
+    @api.constrains('pit_base_amount')
+    def _pit_base_amount_check(self):
+        if not self.pit_base_amount > 0:
```
Comment (elicoidal) [inline l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py]: not sure about this constraint. I think it should >= not only >.
Besides if you have a check you should have as well the following ones: <= wage
Last but not least SI base should be checked as well.

**Comment by elicoidal** [comment]
@xie8899 You need to create a Company Contribution Register and link all company contributions to this register.

File: l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py (hunk 1 above)
Comment (xie8899) [inline l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py:27]: This field is useful. 
for example:
amount_python_compute ： result = ((min(BFPIT,contract.pit_base_amount) - contract.pit_exemption_amount) \* 0.03)

Comment (elicoidal) [inline l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py:27]: OK

File: l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py (hunk 2 above)
Comment (xie8899) [inline l10n_cn_hr_payroll/models/l10n_cn_hr_payroll.py]: `>0` is ok

**Comment by elicoidal** [comment]
:+1: 
@duanyp1991 @pedrobaeza could we square this.
It is been used in production in our company for several months now

**Comment by martintorresvm** [comment]
@pedrobaeza @dreispt @moylop260 Could you review this PR?

cc @elicoidal

**Comment by elicoidal** [comment]
@victormartinelicocorp please review it but in general this is very specific to China so I dont think Pedro or Daniel can help

**Comment by martintorresvm** [comment]
> @victormartinelicocorp please review it but in general this is very specific to China so I dont think Pedro or Daniel can help

I tried but for me Chinese law is also a handicap. Also I have some questions.

Technical speaking nothing to object. 

IMO Functional Test is the most important here.

PEP8 and UT would be nice see if pass but Travis is not available on this repository. Also Runbot and Coveralls.

For the others reviewers a reason why setup all this rules and calculations by data files instead of leave everyone setup on DB by their own. 

If these rules are suitable for every company in China we could try to give as well on the RST document or give some reference to check if it's true even if it's Chinese language

Example Rule Number 1: Here calculate Payslip for employees with Chinese nationality. 21.5 is the number of workable days every month could have is multiplied to base salary...etc. 

What do you think @elicoidal?

**Comment by elicoidal** [comment]
@victormartinelicocorp Can you check with @noahzaozao for the travis setup and create the corresponding PR to fix all missing tools?
As for the functional part, it has been in production for some month now: we will have to improve it with a new version and migrate to v10 soon.
So funcional test is OK for me
Later on we will create better rst README when it is stable.
Let's first merge this

**Pull request merged**